sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -132,6 +141,53 @@ def get_historical_features(
         return job
 
 
+def _assert_expected_columns_in_dataframe(
+    join_keys: Set[str], entity_df_event_timestamp_col: str, entity_df: pandas.DataFrame
+):
+    entity_df_columns = set(entity_df.columns.values)
+    expected_columns = join_keys.copy()
+    expected_columns.add(entity_df_event_timestamp_col)
+
+    missing_keys = expected_columns - entity_df_columns
+
+    if len(missing_keys) != 0:
+        raise ValueError(
```

Comment:
@achals would love to get your take on whether we should be using a custom error type here (or ValueError)? I've accepted #1615, but now that I think about it we probably need tests that cover this failure case. We can always leave those tests as a `TODO` in the code until we have completed our test case refactor, but at that point we probably want different offline stores to raise identical exceptions.

Reply:
I think a specific validation error is better to throw since they can be caught and handled better potentially. 

And yeah, re: tests I'll fill in the gaps in the stuff I just merged.